cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_test.go

```diff
@@ -155,23 +158,26 @@ func TestGetAvailablePackagesStatus(t *testing.T) {
 
 }
 
-func packageFromSpec(spec map[string]interface{}) *unstructured.Unstructured {
+func packageFromSpec(version interface{}, spec map[string]interface{}) *unstructured.Unstructured {
 	return &unstructured.Unstructured{
 		Object: map[string]interface{}{
 			"apiVersion": fmt.Sprintf("%s/%s", packageGroup, packageVersion),
-			"kind":       "Package",
+			"kind":       packageResource,
 			"metadata": map[string]interface{}{
-				"name": fmt.Sprintf("%s.%s", spec["publicName"], spec["version"]),
+				"name": fmt.Sprintf("%s.%s", spec["packageRef"].(map[string]interface{})["refName"], version),
```

Comment:
Hmm, this could panic now right? I see from the way this function is used in setting up test data in structs, that we don't want to return an error here (it'd make it painful to use), but what we can do is update the function to accept a `t *testing.T` as the first arge, then rather than assuming in the line above that the conversion is successful, use the "comma, ok" idiom (see https://golang.org/doc/effective_go#interface_conversions ) to avoid an unnecessary panic in the test code, with something like:

```
pkgRef, ok := spec["packageREf"].(map[string]interface{})
if !ok {
    t.Fatalf("unable to convert %+v to a map[string]interface{}", spec["packageRef"])
}
```

It's a little more verbose, but just good to avoid situations that can panic generally.

Reply:
Thanks for the tip here! I assumed that, given that we control the tests, these situations wouldn't appear. But, yep, you're right, we better avoid these panic situations,